api/ChattO/Application/Payment/Commands/CreateBilling.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Application.Abstractions;
+using Application.Helpers;
+using Application.Payment.DTOs;
+using Domain.Models;
+using FluentValidation;
+using MediatR;
+
+namespace Application.Payment.Commands;
+
+public class CreateBilling
+{
+    public class Command : IRequest<Result<CreateOrderResponse>>
+    {
+        public Guid BillingInfoId { get; set; }
+
+        public Guid OrganizationId { get; set; }
+    }
+
+    public class CommandValidator : AbstractValidator<Command>
+    {
+        public CommandValidator()
+        {
+            RuleFor(x => x.BillingInfoId).NotEmpty();
+            RuleFor(x => x.OrganizationId).NotEmpty();
+        }
+    }
+
+    public class Handler : IRequestHandler<Command, Result<CreateOrderResponse>>
+    {
+        private readonly IBillingService _billingService;
+
+        private readonly IRepository<Billing> _billingRepository;
+
+        private readonly IRepository<BillingInfo> _billingInfoRepository;
+
+        private readonly IValidator<Command> _validator;
+
+        public Handler(IBillingService billingService, IRepository<Billing> repository, IRepository<BillingInfo> billingInfoRepository, IValidator<Command> validator)
+        {
+            _billingService = billingService;
+            _billingRepository = repository;
+            _billingInfoRepository = billingInfoRepository;
+            _validator = validator;
+        }
+
+        public async Task<Result<CreateOrderResponse>> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var validationResult = await _validator.ValidateAsync(request);
+            if (!validationResult.IsValid)
+            {
+                return Result.Failure<CreateOrderResponse>(validationResult.ToString(" "));
+            }
+
+            var billingId = Guid.NewGuid();
+            var billing = new Billing() 
```

Comment:
Here you can also do mapping between Command and Billing

Reply:
done